### ISSUE TITLE
Cluster-essentials size capacity

### ISSUE THREAD

**Issue by Demonsthere** [kyma-project/kyma#9454]
**Description**

When adding new CRDS into cluster-essentials, I came across this error:

```
2020/09/07 08:44:53 Step error:  Details: Helm installation of release "cluster-essentials" failed: create: failed to create: Secret "sh.helm.release.v1.cluster-essentials.v1" is invalid: data: Too long: must have at most 1048576 characters
```

The size of the chars (including CRDs in the files folder) exceeds the maximum size that helm can handle when using k8s secrets as its backend driver.

**Expected result**

The chart is installed, regardless of the number of CRDs we want to install

**Actual result**

We have a strong dependency on the size/number of CRDs in the chart, and using the current logic, would need to split into cluster-essentials-1 and cluster-essentials-2 which again, does not scale.

**Steps to reproduce**

Add mode CRDs to the files folder, at some point we will see this error (I was adding cert-manager CRDs, 5 more files)

**Troubleshooting**

We need to think and implement a new method of installing CRDs, some propositions:
- using the helm crd folder method - investigate if crds are part of the release 
- store the files in a remote folder/server (github, gcp) and install from there - dependency on network and some service
- create an image with the crds and install locally

**Comment by Demonsthere** [comment]
Update:
the `crds` folder methods of installation is easy and fast, but the crds are still loaded into the `sh.helm.release` secrets.

**Comment by Demonsthere** [comment]
https://github.com/kyma-project/kyma/pull/9362 By merging this and removing knative-serving, we may thin cluster-essentials enough to be able to install cert-manager crds. This however is only delaying the problem and not a proper solution

**Comment by Demonsthere** [comment]
Installation using a dedicated image: https://github.com/kyma-project/kyma/pull/9455

**Comment by piotrmsc** [comment]
I also agree that creating another helm release like `cluster-essentials-2` is not good idea.
I think we should check crds folder in helm3 and see if it's going to be part of release. 

As an alternative approach, we could install CRDs separately, before installing any helm release, therefore making it completely independent from helm management system. Here we could : 
- Install CRD as some kind of init step in the CLI or 
- Install CRD as init step in the kyma-operator

Personally, I think installing CRDs first in the CLI while invoking `kyma install/upgrade` is better than embedding it in the kyma-operator. CLI supports installation from local source as well from release so CRDs should have the same support I CLI therefore still be part of github repo or moved to bucket etc.

Installing CRDs in the operator also is in general not bad idea, a good example could be istio here:
Istio is installing CRDs via istioctl or via istio operator. Both CLI and operator share libs for that. However, there's a major difference between our operator and istio operator/cli : 
- We are a generic operator and our component list changes, plus you can install charts from remote repo dynamically. Between releases, new components like cert-manager can be added therefore the CRD list is not constant.
- Istio operator / istioctl has single responsibility -> install/upgrade only istio related components. 

Also, we have pushed our installation process to use CLI first so it is a prerequisite to have it. That's why I think the installation of CRDs could be there. 
Additionally, if we take a look on other k8s ecosystems like cert-manager or helm-operator, They are not self-contained, meaning you also have a pre step : install crds and in my opinion that's a good approach.

If helm crds folder will not give us the expected results I would propose moving out crds to CLI. 
I'm not describing here technical details like one CLI command or 2 separate ones as that will be subject for a detailed proposal once any idea will be picked.

**Comment by piotrmsc** [comment]
Update from my side :  I also confirm that having crds in `crds` folder in helm3 still makes it part of the release, however, I was able to install all current cards and cert manager crds this way but in general it does not solve the issue.

**Comment by piotrmsc** [comment]
We should do the spike where we will find answers to the following questions : 
- Where to store CRDs : in a bucket / still on github in the resources / on GitHub as release artifacts / somewhere else
- What implications all of the possibilities (one by one) have to our current installation/upgrade process including pipelines and release. What would need to be changed and where.

Key assumptions : 
- CRDs should be installed also from sources to match also the option of installing kyma from sources
- To do not brake the contract with upgrade it should work the same way as currently with cluster-essential - be re-applied.
- Installation CRD should be also part of the rest of CRDs, should not be treated as a separate CRD


In the end we should also have a small poc implementation in the CLI that will show us kyma components and crds installed from sources.
